### PR TITLE
python312Packages.locationsharinglib: drop nose dependency; modernize

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Part of #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/locationsharinglib/default.nix [hunk 1]
```diff
@@ -1,65 +1,53 @@
...
   requests,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "locationsharinglib";
   version = "5.0.3";
-  format = "setuptools";
-
-  disabled = pythonOlder "3.7";
```
Comment (fabaff) [inline pkgs/development/python-modules/locationsharinglib/default.nix:20]: Should stay and be update to 3.9, see https://github.com/costastf/locationsharinglib/blob/5.0.3/setup.py#L49

Comment (pyrox0) [reply pkgs/development/python-modules/locationsharinglib/default.nix:20]: Fixed

File: pkgs/development/python-modules/locationsharinglib/default.nix [hunk 2]
```diff
@@ -1,65 +1,56 @@
...
   postPatch = ''
     # Tests requirements want to pull in multiple modules which we don't need
     substituteInPlace setup.py \
-      --replace "tests_require=test_requirements" "tests_require=[]"
+      --replace-fail "tests_require=test_requirements" "tests_require=[]"
     substituteInPlace requirements.txt \
-      --replace "coloredlogs>=15.0.1" "coloredlogs" \
-      --replace "pytz>=2023.3" "pytz"
+      --replace-fail "coloredlogs>=15.0.1" "coloredlogs" \
+      --replace-fail "pytz>=2023.3.post1" "pytz"
```
Comment (emilazy) [inline pkgs/development/python-modules/locationsharinglib/default.nix:31]: Just checking if `pythonRelaxDeps`/`pythonRemoveDeps` might work here?

Comment (pyrox0) [reply pkgs/development/python-modules/locationsharinglib/default.nix:31]: Just tested, it seems to work without this patch, so I've removed it.

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 330786` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.locationsharinglib</li>
    <li>python311Packages.locationsharinglib.dist</li>
    <li>python312Packages.locationsharinglib</li>
    <li>python312Packages.locationsharinglib.dist</li>
  </ul>
</details>

**Pull request merged**